test/test_cli_integration.py

```diff
@@ -71,6 +71,28 @@ def replace_policy(self, policy_path):
         return invoke_cli(self, self.cli_auth_params,
             ['policy', 'replace', 'root', policy_path])
 
+    def replace_policy_from_string(self, policy):
+        output = None
+        with tempfile.NamedTemporaryFile() as temp_policy_file:
+            temp_policy_file.write(policy.encode('utf-8'))
+            temp_policy_file.flush()
+
+            # Run the new apply that should not result in newly created roles
+            output = self.replace_policy(temp_policy_file.name)
+
+        return output
+
+    def apply_policy_from_string(self, policy):
```

Comment:
Step in the right direction, though I think this could be DRYed up even more.

Reply:
Yeah this needs some DRYing up for sure as the tests are getting bulkier. I might defer this to next piece of work on this file so that we can get this into PyPI.